File: CSC454HW5/CppImpOfHW4/EventModelInterface.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   EventModelInterface.cpp
 * Author: chrisrk192
 * 
 * Created on November 7, 2017, 2:16 PM
 */
#include <string>
#include <vector>
#include "EventModelInterface.h"

EventModelInterface::EventModelInterface() {
}

EventModelInterface::EventModelInterface(const EventModelInterface& orig) {
}

EventModelInterface::~EventModelInterface() {
}

std::vector<std::string> EventModelInterface::OutPut(){

}   

void EventModelInterface::StateTransitionExt(std::vector<std::string> inputItems, double e){

}

void EventModelInterface::StateTransitionInt(){

}


void EventModelInterface::StateTransitionCon(std::vector<std::string> inputItems){

}

double EventModelInterface::TimeAdvance(){
}


std::string EventModelInterface::GetStatus(){

}

File: CSC454HW6/FinalVer/javaEvenBetter/CSC454HW6/src/csc454hw6/EventModelInterface.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package csc454hw6;

/**
 *
 * @author chrisrk192
 * Interface all models used in the simulation framework or eventNetwork
 */
public interface EventModelInterface<I, O>{
    O OutPut();    
    void StateTransitionExt(I inputItems, double e) throws CantPerformAction;
    void StateTransitionInt() throws CantPerformAction;
    void StateTransitionCon(I inputItems) throws CantPerformAction;
    double TimeAdvance();
    public String GetStatus();
}
class CantPerformAction extends Exception {
    public CantPerformAction(String message){
        super(message);
    }
}